geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/DaoSpatialite.java

```diff
@@ -436,9 +439,51 @@ public static void updateFeatureGeometry(String id, Geometry geometry, String ge
         sbIn.append(SpatialiteUtilities.SPATIALTABLE_ID_FIELD).append("=");
         sbIn.append(id);
         String insertQuery = sbIn.toString();
-        database.exec(insertQuery, null);
+        dbSpatialite.exec(insertQuery, null);
+    }
+    public static double[] longLat2Srid(Database dbSpatialite, double lon, double lat, String s_srid)  throws Exception {
```

Comment:
But the srid of the map can only be 3857, right? Why is this necessary?
I would love to see generic srids involved, but libraries that start from OSM tiles only support 3857